vertx-mysql-client/src/main/java/io/micronaut/configuration/vertx/mysql/client/condition/RequiresVertxMySQLClient.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2017-2018 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package io.micronaut.configuration.vertx.mysql.client.condition;
+
+import io.micronaut.configuration.vertx.mysql.client.MySQLConnectionSettings;
+import io.micronaut.context.annotation.Requires;
+import io.vertx.mysqlclient.MySQLConnectOptions;
+import io.vertx.sqlclient.PoolOptions;
+
+import java.lang.annotation.*;
+
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PACKAGE, ElementType.TYPE})
+@Requires(property = MySQLConnectionSettings.PREFIX)
```

Comment:
I'm not sure whether we need both `MySQLConnectionSettings` and `MySQLPoolSettings`, is `MySQLClientSettings` enough here?

Reply:
I don't think it's a problem, since it's separate in verx-sql-client

Reply:
IMHO It is more like a configuration namespace here, I think it would be better to keep only one for each project and we have already two separated options object anyway.

Reply:
I've changed it. your are right

Reply:
I think `MySQLClientConfiguration` and `MySQLClientSettings` is better.

Reply:
Thank you for your advice. Naming is really too difficult. I have already submitted it.